src/util/alloc/malloc_allocator.rs

```diff
@@ -1,28 +1,31 @@
+use std::sync::Arc;
+
 use crate::policy::marksweepspace::malloc_ms::MallocSpace;
 use crate::policy::space::Space;
 use crate::util::alloc::Allocator;
 use crate::util::opaque_pointer::*;
 use crate::util::Address;
 use crate::vm::VMBinding;
-use crate::Plan;
+
+use super::allocator::AllocatorContext;
 
 #[repr(C)]
 pub struct MallocAllocator<VM: VMBinding> {
     /// [`VMThread`] associated with this allocator instance
     pub tls: VMThread,
     /// [`Space`](src/policy/space/Space) instance associated with this allocator instance.
     space: &'static MallocSpace<VM>,
-    /// [`Plan`] instance that this allocator instance is associated with.
-    plan: &'static dyn Plan<VM = VM>,
+    context: Arc<AllocatorContext<VM>>,
+    _pad: usize,
```

Comment:
I am currently use a padding field to make sure the size of the allocator types stays the same as before. I plan to remove this field, and update the bindings.